json/src/main/java/com/esri/json/hadoop/UnenclosedJsonRecordReader.java

```diff
@@ -110,37 +138,28 @@ private boolean moveToRecordStart() throws JsonParseException, IOException {
 			inputReader.mark(100);
 			
 			// ok last char was '{', skip till we get to a '"'
-			while (true) {
-				next = inputReader.read(); readerPosition++;
-				
-				// end of stream, no good
-				if (next < 0) {
-					return false;
-				}
-				
-				if (!Character.isWhitespace((char)next)) {
-					break;
-				}
+			next = getNonWhite();
+			if (next < 0) {   // end of stream, no good
+				return false;
 			}
-			
 			if (next != '"') {
 				continue;
 			}
-			
+
+			boolean inEscape = false;
 			String fieldName = "";
 			// next should be a field name of attributes or geometry
-			while (true) {
-				next = inputReader.read(); readerPosition++;
-				
-				// end of stream, no good
-				if (next < 0) {
+			while (next != '{') {  // If opening brace, the previous one was quoted
```

Comment:
Opening brace in quoted string literal - tested with escapes, but probably not inherently related to escapes.  Tested by `TestEscOpenLast` with `unenclosed-json-esc5.json`.